### PR TITLE
basic hypverv machine implementation

### PR DESCRIPTION
**Pull request by baude**
with libhvee, we are able to do the basics of podman machine management on hyperv.  The basic functions like init, rm, stop, and start are all functional.  Start and stop will periodically throw a benign error processing the hyperv message being returned from the action.  The error is described in the todo's below.

notable items:

* no podman commands will work (like ps, images, etc)
* the machine must be initialized with --image-path and fed a custom image.
* disk size is set to 100GB statically.
* the vm joins the default hyperv network which is TCP/IP network based.
* podman machine ssh does not work
* podman machine set does not work
* you can grab the ip address from hyperv and fake a machine connection with `podman system connection`.
* when booting, use the hyperv console to know the boot is complete.

TODOs:
* podman machine ssh
* podman machine set
* podman machine rm needs force bool
* disk size in NewMachine is set to 100GB
* podman start needs to wait until fully booted
* establish a boot complete signal from guest
* implement gvproxy like user networking
* fix benign failures in stop/start -> Error: error 2147749890 (FormatMessage failed with: The system cannot find message text for message number 0x%1 in the message file for %2.)

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
none
```

### CODE REVIEW

**Comment by baude** [comment]
@n1hility ptal as well

File: pkg/machine/hyperv/config.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+		return false, "", err
+	}
+	for _, vm := range vms {
+		foo := vm.IsStarting()
+		_ = foo
+		bar := vm.State() == hypervctl.Enabled
+		_ = bar
+		if vm.IsStarting() || vm.State() == hypervctl.Enabled {
+			return true, vm.ElementName, nil
+		}
```
Comment (ashley-cui) [inline pkg/machine/hyperv/config.go:45]:
```suggestion
		if vm.IsStarting() || vm.State() == hypervctl.Enabled {
			return true, vm.ElementName, nil
		}
```

**Review by ashley-cui — COMMENTED** [review]
Starting with these comments, continuing to review

File: pkg/machine/pull.go [hunk 2]
```diff
@@ -280,6 +288,34 @@ func decompressEverythingElse(src string, output io.WriteCloser) error {
 	return err
 }
 
+func decompressZip(src string, output io.WriteCloser) error {
+	tar, err := exec.LookPath("tar")
+	if err != nil {
+		return errors.New("unable to find tar binary to unzip image")
+	}
+	cmd := exec.Command(tar, "-x", "-O", "-f", src)
```
[the remark this answers is not in the document]
Comment (baude) [reply pkg/machine/pull.go:296]: it does on windows

[the remark this answers is not in the document]
Comment (baude) [reply pkg/machine/pull.go:296]: ok, good poiknt though, i should probably prevent things going into conditional if not windows.  thx!

File: pkg/machine/hyperv/config.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+	}
+
+	// Nuke the config and dataDirs
+	if err := os.RemoveAll(configDir); err != nil {
+		prevErr = handlePrevError(err, prevErr)
+	}
+	if err := os.RemoveAll(dataDir); err != nil {
+		prevErr = handlePrevError(err, prevErr)
+	}
```
Comment (ashley-cui) [inline pkg/machine/hyperv/config.go:237]: Does removing configDir remove just this Machine's config, or all machines?

Comment (baude) [reply pkg/machine/hyperv/config.go:237]: all

**Pull request merged**